### PR TITLE
Remove dependency on JetBrains.Annotations.

### PR DESCRIPTION
**Pull request by ejball**
Use source code copied from ReSharper » Options » Code Inspection » Code Annotations. (This allows the annotations to be seen by ReSharper where the library is used.)

### CODE REVIEW

File: src/Faithlife.Utility/JetBrainsAnnotations.cs [hunk 1]
```diff
@@ -0,0 +1,1237 @@
+// <auto-generated>From ReSharper » Options » Code Inspection » Code Annotations</auto-generated>
```
Comment (bgrainger) [inline src/Faithlife.Utility/JetBrainsAnnotations.cs:1]: Would it be worth naming this file `JetBrainsAnnotations.g.cs`?

(Are there some tools that are sensitive to that filename pattern and don't "inspect" it, etc.? I can't remember.)

Comment (ejball) [reply src/Faithlife.Utility/JetBrainsAnnotations.cs]: Good idea. Done.

**Review by bgrainger — APPROVED** [review]

**Comment by ddunkin** [comment]
Why did you choose this over defining `JETBRAINS_ANNOTATIONS` to compile the annotations into the assembly?

**Comment by ejball** [comment]
Mostly I don't like forcing an otherwise unneeded dependency on clients of a utility library.

It can also cause annoying build warnings if the version we use doesn't match the version they use.

But that's probably a better strategy for less utilitarian libraries. And perhaps you could persuade me that it's for the best here as well.

**Pull request merged**